sympy/polys/tests/test_hypothesis.py

```diff
@@ -1,19 +1,18 @@
 from hypothesis import given
 from hypothesis import strategies as st
-from sympy.abc import x
-from sympy.polys.polytools import Poly
+from sympy.testing.hypothesis import polynomial
+from sympy import gcd, lcm
+
 
 
 @given(
-    coefficients1=st.lists(st.integers()),
-    coefficients2=st.lists(st.integers()),
-    coefficients3=st.lists(st.integers()),
+    f=polynomial(),
+    g=polynomial(),
+    r=polynomial(),
+    x=st.integers(),
+    y=st.integers(),
```

Comment:
We should only test polynomials here. We can test integer gcd (`igcd`) in a separate test.